developers/release.md

```diff
@@ -0,0 +1,348 @@
+
+```
+cd katello-misc/repos
+export VER=X.Y
+sed -i "s#/nightly/#/$VER/#g" katello*.repo
+git commit -a -m 'Updating repository URL to X.Y'
+tito tag
+tito release koji
+/usr/local/bin/katello-mash-split-X.Y.py
+```
+
+### Step 11: Release X.Y Release Candidate RPMs
+
+For this step we'll need to update the Jenkins job that pushes RPMs to fedorapeople and run the job. Go to `http://ci.theforeman.org/view/Katello%20Pipeline/job/release_push_rpm_katello/configure` and add X.Y to the list of RELEASE choices and save the configuration. Now head to `http://ci.theforeman.org/view/Katello%20Pipeline/job/release_push_rpm_katello/build?delay=0sec`, choose the X.Y version you just added and click `Build`. This will run a script that copies the RPMs from Koji to fedorapeople.
+
+### Step 12: Update katello-deploy
+
+We maintain a deployment tool, katello-deploy, for easily testing and deploying Katello versions. You'll want to clone the repository (https://github.com/Katello/katello-deploy.git) and update the following:
+
+1. Edit the README to indicate the new version and the OSes it works with.
+1. Update the foreman version matcher - https://github.com/Katello/katello-deploy/blob/master/setup.rb#L10
+1. Add new boxes for X.Y - https://github.com/Katello/katello-deploy/blob/master/Vagrantfile#L28
+
+### Step 13: Test
+
+Use the newly updated katello-deploy to spin up an X.Y box and test for the following:
+
+* Installation
+* UI
+* Repository sync and content view publish
+* Can install katello-agent for el5,6, 7 from the client repo (without the need of the main katello repo enabled)
+* Run a full bats test with content
+
+### Step 14: Announce RC
+
+Now you are ready to announce the RC by sending an email to the foreman-users and foreman-dev mailing lists pointing users at the installation instructions. You should also open a PR to `katello.org` to update the news section with an announcement of the RC release.
+
+### Step 15: Fix Bugs
+
+Start off by creating a label that represents the X.Y version, e.g. `KATELLO-2.1` on the relevant repositories. The workflow until release will be as follows: 
+
+1. Mark any open pull request that should go into X.Y with 'Needs Backporting' and 'Katello X.Y'.
+1. Periodically go through and backport all the merged 'needs backporting' PRs to the X.Y branch and open one new PR against the X.Y branch with all the backports
+1. Remove the 'needs backporting' label from the PRs after cherry picking
+1. After generated 'backport' is merged, tag and release, mash, and copy-repos to fedorahosted
+
+### Step 16: Release!
+
+After iterating through Step 15 a few times over the course of roughly a month, and in sync with Foreman's release date decide that you are ready to release.
+
+```
+cd katello-misc/repos
+sed -i "s#/gpgcheck=0/#/gpgcheck=1/#g" katello*.repo
+git commit -a -m 'Turning on GPG key checking for X.Y'
+tito tag
+tito release koji
+```
+
+##### Download Unsigned RPMs
```

Comment:
@jlsherrill can you please review from this point down related to signing RPMs? I took this mostly verbatim from the old fedorahosted release page.